packages/toolshed/src/core/collectors.ts

```diff
@@ -0,0 +1,84 @@
+import { BytesLike, ethers, HDNodeWallet, Signature } from 'ethers'
+
+import { IGraphTallyCollector } from '@graphprotocol/subgraph-service'
+
+/**
+ * Generates a signed RAV calldata
+ * @param graphTallyCollector The Graph Tally Collector contract
+ * @param signer The signer
+ * @param collectionId The collection ID
+ * @param payer The payer
+ * @param serviceProvider The service provider
+ * @param dataService The data service
+ * @param timestampNs The timestamp in nanoseconds
+ * @param valueAggregate The value aggregate
+ * @param metadata The metadata
+ * @returns The encoded signed RAV calldata
+ */
+export async function getSignedRAVCalldata(
+  graphTallyCollector: IGraphTallyCollector,
+  signer: HDNodeWallet,
+  allocationId: string,
+  payer: string,
+  serviceProvider: string,
+  dataService: string,
+  timestampNs: number,
+  valueAggregate: bigint,
+  metadata: BytesLike
+) {
+  const ravData = {
+    collectionId: ethers.zeroPadValue(allocationId, 32),
+    payer: payer,
+    serviceProvider: serviceProvider,
+    dataService: dataService,
+    timestampNs: timestampNs,
+    valueAggregate: valueAggregate,
+    metadata: metadata
+  }
+
+  const encodedRAV = await graphTallyCollector.encodeRAV(ravData)
```

Comment:
Same as before, I think if we can remove the need for a contract interaction here it's better.